apacheconfig/lexer.py

```diff
@@ -220,16 +232,23 @@ def t_multiline_OPTION_AND_VALUE(self, t):
         t.lexer.begin('INITIAL')
 
         value = t.lexer.lexdata[t.lexer.code_start:t.lexer.lexpos + 1]
-        t.lexer.lineno += len(re.findall(r'\r\n|\n|\r', value))
-        value = value.replace('\\\n', '').replace('\r', '').replace('\n', '')
 
-        option, value = self._parse_option_value(value)
+        # don't eat trailing whitespace
+        # TODO(sydli) fix escaped newline case
+        any_trailing_whitespace = [m.start(0) for m in re.finditer(r'[ \t\r\n]+$', value)]
```

Comment:
I still can't wrap my head around this line. Are we operating on a block of text which potentially can have many lines ending with spaces? I wonder why we are looking at exactly the first one? Sorry for likely silly questions!

Reply:
I changed this around a bit to make the purpose clearer, and will definitely add a more in-depth comment :)
```
# Since the lexer needs to be able to tokenize whitespace, rewind
# lexpos to before this token's trailing whitespace.
stripped_value = value.rstrip()
if stripped_value != value:
  value = value[:len(stripped_value)]
```

It could also be possible to do this by changing around the lexing regexes for the `multiline` state, but I found this slightly easier to work with for the time being.